File: examples/get_all.js
"use strict";

var botnana = require('../index');

function get_all() {
    // Show reponse data
    botnana.debug_level = 1;
    botnana.once("ready", function() {
        for (var i = 1; i <= botnana.ethercat.slave_count; i = i + 1) {
            botnana.ethercat.slave(i).get();
        }
    });

    botnana.start("ws://192.168.7.2:3012");
 }

get_all();

setTimeout(function() {
    botnana.ethercat.slave(1).get_diff();
}, 2000)

File: botnanars/examples/get_all.rs
extern crate botnanars;
use botnanars::Botnana;
use std::{thread, time};
use std::sync::{Arc, Mutex};

fn main() {
    let botnana = Arc::new(Mutex::new(Botnana::new()));

    let btn = botnana.clone();

    botnana.lock().unwrap().once("ready", move |_| {
        let btn = btn.clone();
        thread::spawn(move || {
            let slave_count = btn.lock().unwrap().ethercat.get_slaves_count();
            for i in 1..slave_count {
                match btn.lock().unwrap().ethercat.slave(i) {
                    Some(s) => s.get(),
                    None => {
                        println!("get None");
                    }
                }
            }

            loop {
                for i in 1..slave_count {
                    match btn.lock().unwrap().ethercat.slave(i) {
                        Some(s) => s.get_diff(),
                        None => {
                            println!("get None");
                        }
                    }
                }
                thread::sleep(time::Duration::from_millis(2000));
            }
        });
    });

    botnana.lock().unwrap().start("ws://192.168.50.222:3012");

    loop {}
}
